examples/weather_station/src/app_task.cpp

```diff
@@ -254,6 +277,47 @@ void AppTask::MeasurementsTimerHandler()
 	sAppTask.UpdateClusterState();
 }
 
+void AppTask::OnIdentifyStart(Identify *)
+{
+	uint16_t identifyTimeS;
+	EmberAfStatus status = emberAfReadAttribute(kIdentifyEndpointId, ZCL_IDENTIFY_CLUSTER_ID, ZCL_IDENTIFY_TIME_ATTRIBUTE_ID, CLUSTER_MASK_SERVER, reinterpret_cast<uint8_t *>(&identifyTimeS), sizeof(identifyTimeS), NULL);
+	if (status != EMBER_ZCL_STATUS_SUCCESS) {
+		LOG_ERR("Reading identify time failed %x", status);
+		return;
+	}
+	sIdentifyFullDurationTimeMs = identifyTimeS * 1000;
+	k_timer_start(&sIdentifyTimer, K_MSEC(kIdentifyIntervalMs), K_MSEC(kIdentifyIntervalMs));
+}
+
+void AppTask::OnIdentifyStop(Identify *)
+{
+    // empty
+}
+
+void AppTask::IdentifyTimerHandler()
+{
+	sIdentifyDurationMs += kIdentifyIntervalMs;
+	/* Every second decrement IdentifyTime attribute value and update cluster state */
+	if (sIdentifyDurationMs % kIdentifyDecrementIntervalMs == 0) {
+		EmberAfStatus status;
+		int timeLeft = sIdentifyFullDurationTimeMs - sIdentifyDurationMs;
+		uint16_t identifyTime = timeLeft > 0 ? timeLeft : 0;
+		status = emberAfWriteAttribute(kIdentifyEndpointId, ZCL_IDENTIFY_CLUSTER_ID,
+					       ZCL_IDENTIFY_TIME_ATTRIBUTE_ID, CLUSTER_MASK_SERVER,
+					       reinterpret_cast<uint8_t *>(&identifyTime), ZCL_INT16U_ATTRIBUTE_TYPE);
+
+		if (status != EMBER_ZCL_STATUS_SUCCESS) {
+			LOG_ERR("Updating identify time failed %x", status);
+		}
+
+		if (identifyTime == 0) {
+			k_timer_stop(&sIdentifyTimer);
```

Comment:
I think `OnIdentifyStop` should call that and this function should simply `toggle the buzzer state`. Wouldn't it be enough?

Reply:
> I think OnIdentifyStop should call that and this function

Right, done

> this function should simply toggle the buzzer state. Wouldn't it be enough?

What do you mean by this function? `OnIdentifyStop` or the handler? Anyway buzzer needs to be toggled in the handler, as it beeps in the 500 ms on - 500 ms off cycle for the identify time.

Reply:
I mean that the timer handler should just toggle the buzzer and it shouldn't care if the indentify is enabled. `OnIdentifyStop` will reset the timer when the handler should no longer be called.

Reply:
@Damian-Nordic yeah, I tested that indeed cluster state is updated automatically by the core, so removed all of the code handling it.